docs/features/source-generators.md

```diff
@@ -0,0 +1,285 @@
+# Source Generators
+
+
+## Summary
+
+Source generators aim to enable _compile time metaprogramming_, that is, code that can be created
+at compile time and added to the compilation. Source generators will be able to read the contents of 
+the compilation before running, as well as access any _additional files_, enabling generators to
+introspect both user C# code and generator specific files.
+
+> **Note**: This proposal is separate from the [previous generator design](generators.md)
```

Comment:
💡 It would be good to also update the other page to mention its relation to this one.